### PR TITLE
Implementing skip property

### LINKED ISSUE #104: feature: skip rule based on conditional
An alternative to jumping (next).

### Supported inputs

**Function**
```javascript
const rule = {
  skip: (output, rule, bot) => output.personType === 'Individual',
}
```

**Definition** (suggested by @cristianossd)
```javascript
bot.XYZ.define('haveEmail', (store) => !!store.email);
```
```yaml
- skip: haveEmail
```

**Boolean**
```yaml
- skip: true|false
```

### PR DESCRIPTION
**Pull request by cristianossd**
This approach uses bot actions to execute the skip function. My suggestion is not use the boolean value directly in skip because, if I want to pass the boolean in skip rule, I just remove the rule. What do you think @andersonba ?

It closes #104

### CODE REVIEW

**Comment by cristianossd** [comment]
I updated with your initial approach, @andersonba. I agree with u that it makes more sense in terms of bot structure

File: docs/docs/rule.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ You can define a rule using two styles
 | actions | `[]` | array[[RuleAction](#ruleaction)] | Executes an action after bot's message
 | preActions | `[]` | array[[RuleAction](#ruleaction)] | Executes an action before bot's message
 | postActions | `[]` | array[[RuleAction](#ruleaction)] | Executes an action after user's answer
+| skip | - | RuleTypeSkip | Skip rule based on conditional function
```
Comment (andersonba) [inline docs/docs/rule.md]: 1. change default to `false`
1. in type, put `boolean | (output, rule, bot) => boolean`

File: src/core/sanitizers.ts [hunk 2]
```diff
@@ -32,9 +32,16 @@ export function sanitizeRule(input: IRule): IRule {
...
     return o;
-  });
+    });
+
   rule.passive = rule.passive === undefined ? true : rule.passive;
 
+  if (typeof rule.skip === 'undefined') {
+    rule.skip = () => false;
+  } else {
+    rule.skip = typeof rule.skip === 'boolean' ? () => !!rule.skip : rule.skip;
```
Comment (andersonba) [inline src/core/sanitizers.ts]:
```javascript
rule.skip = typeof rule.skip === 'function' ? rule.skip : () => !!rule.skip;
```

To ensure cases like that:
`skip: blah blah`

File: src/types.ts [hunk 3]
```diff
@@ -61,6 +62,10 @@ export type RuleTypeTransform = (
   (value: any, rule?: IRule, bot?: YveBot) => Promise<any>
 );
 
+export type RuleTypeSkip = (
+  (output: any, rule?: IRule, bot?: YveBot) => boolean
```
Comment (andersonba) [inline src/types.ts]: `output: Object`

**Comment by rafaelverger** [comment]
Nice job, guys!

**Pull request merged**